### ISSUE TITLE
spider: Option to stop spider if site starts blocking requests

### ISSUE THREAD

**Issue by double16** [zaproxy/zaproxy#8170]
### Is your feature request related to a problem? Please describe.

Site protections may detect the spider and block it. Or, the server(s) may become overloaded and return 500, 502, etc.

### Describe the solution you'd like

Spider can be configured with conditions to stop. HTTP error codes, page regex. It should be tied into the existing Custom Page feature.

### Describe alternatives you've considered

Watch the spider and manually stop it. Perhaps a script?

### Would you like to help fix this issue?

- [X] Yes

**Comment by psiinon** [comment]
My worry is that the conditions might end up being very wide ranging. 
I think a script could be a more flexible alternative..

**Comment by thc202** [comment]
Agreed, this should be handled generally, it's not just the spider that has this problem.

**Comment by thc202** [comment]
Renamed to be more generic.

See also: https://groups.google.com/g/zaproxy-users/c/8aAtnoU2-Ow/m/lsMOiTczAwAJ

**Comment by psiinon** [comment]
Partly now covered by https://www.zaproxy.org/docs/getting-further/automation/target-scanning-issues/
